executor/adapter.go

```diff
@@ -112,8 +112,20 @@ func colNames2ResultFields(schema *expression.Schema, names []*types.FieldName,
 // The reason we need update is that chunk with 0 rows indicating we already finished current query, we need prepare for
 // next query.
 // If stmt is not nil and chunk with some rows inside, we simply update last query found rows by the number of row in chunk.
-func (a *recordSet) Next(ctx context.Context, req *chunk.Chunk) error {
-	err := Next(ctx, a.executor, req)
+func (a *recordSet) Next(ctx context.Context, req *chunk.Chunk) (err error) {
+	defer func() {
+		r := recover()
+		if r == nil {
+			return
+		}
+		if str, ok := r.(string); !ok || !strings.HasPrefix(str, memory.PanicMemoryExceed) {
+			panic(r)
+		}
```

Comment:
This logic is copied from `conn.go:1235` and `adapter.go:291`.
How about using another PR to solve this problem?

Reply:
The logic here is different from what in `conn.go` or `adapter.go` because each of them has (or maybe has) its confidence that the caller can handle the panic of `memory.PanicMemoryExceed`. But `recordSet` does not have, I think we may not introduce more bug in this PR.